### DISCUSSION TITLE
fetching JSON files from storage

### DISCUSSION THREAD

**Discussion by derFBeste** [supabase/supabase#16472] [Questions] [answered]
Hi,

I'm trying out storage to store JSON files. I'm using supabase-js to retrieve the files in a frontend. In dev tools I can see in the network tab the response is the payload I'm expecting. It's an array of objects. The problem is when I try to use it in the front end the data property is always an empty object.

I'm probably missing something easy.

Here's an abbreviated sample of how I'm using the client:

```js
const { data, error } = await supabase.storage
    .from("my_bucket")
    .download("my_file.json");

console.log(JSON.stringify(data)):
```

The above will log an empty object.

Any ideas?

**Comment by derFBeste** [comment]
Or if someone could point me to some sample code with retrieving JSON from buckets that'd be helpful. All the examples I've found use images.

**Reply by achom4nu** [reply]
Do you still know where to find those examples with images?

**Comment by derFBeste** [comment, marked as answer]
The data is a Blob so must be read. This worked for me:

`const result = await data?.text();`

https://developer.mozilla.org/en-US/docs/Web/API/Blob#extracting_data_from_a_blob 

Firefox automatically reads the file in the network tab which threw me off a little. Chrome doesn't.
